internal/watcher/validator.go

```diff
@@ -100,17 +101,24 @@ func (u *Updatable) Update() error {
 
 	u.log.Infof("Updating expected measurements")
 
-	var measurements map[uint32][]byte
+	var measurements measurements.M
 	if err := u.fileHandler.ReadJSON(filepath.Join(constants.ServiceBasePath, constants.MeasurementsFilename), &measurements); err != nil {
 		return err
 	}
-	u.log.Debugf("New measurements: %v", measurements)
+	u.log.Debugf("New measurements: %+v", measurements)
 
-	var enforced []uint32
-	if err := u.fileHandler.ReadJSON(filepath.Join(constants.ServiceBasePath, constants.EnforcedPCRsFilename), &enforced); err != nil {
-		return err
+	// handle legacy measurement format, where expected measurements and enforced measurements were stored in separate data structures
+	// TODO: remove with v2.4.0
+	if _, err := u.fileHandler.Stat(filepath.Join(constants.ServiceBasePath, constants.EnforcedPCRsFilename)); err == nil {
+		u.log.Debugf("Detected legacy format. Loading enforced PCRs...")
+
+		var enforced []uint32
+		if err := u.fileHandler.ReadJSON(filepath.Join(constants.ServiceBasePath, constants.EnforcedPCRsFilename), &enforced); err != nil {
+			return err
+		}
+		measurements.SetEnforced(enforced)
+		u.log.Debugf("Merged measurements with enforced values: %+v", measurements)
 	}
```

Comment:
```suggestion
	var enforced []uint32
	if err := u.fileHandler.ReadJSON(filepath.Join(constants.ServiceBasePath, constants.EnforcedPCRsFilename), &enforced); err == nil {
		u.log.Debugf("Detected legacy format.")
		measurements.SetEnforced(enforced)
		u.log.Debugf("Merged measurements with enforced values: %+v", measurements)
	} else if !errors.Is(err, fs.ErrNotExist) {
		return err
	}
```
In most cases, avoid TOCTOU-like code and check the error of the actual action instead.